### PR TITLE
Moved app key to plist; added ChangeLog

### PR DESCRIPTION
**Pull request by qinweigong**
@aaustin

### CODE REVIEW

**Comment by aaustin** [comment]
So this is close but it doesn't solve the actual problem. The problem statement is that our current SDK implementation stores the app id in NSUserDefaults, which can be cleared. If a user clears the settings for one of our apps, it will start sending bnc_no_value in app key (and every other field). Does that make sense?

Facebook and other SDKs have the user just put the app key into the plist file, and then the SDK internal retrieve it from the plist automatically. It's never stored in NSUserDefaults. The developer never needs to retrieve it from the plist file themselves. Does that make sense?

Note, this also applies to identity, session and fingerprint ids as well, but it's probably fine to leave at is for now. Eventually, we should be saving those to a file instead of defaults so users can't clear it.

**Comment by qinweigong** [comment]
Oh, got it!
So, I also need to change the getAppKey and setAppKey functions in BNCPreferenceHelper, correct?

**Comment by aaustin** [comment]
Let's make sure to support legacy implementations for now if we can't get
the key from the plist file - so don't gut it

**Comment by qinweigong** [comment]
I have the backward compatibility code in getAppKey. Please review my latest commit.

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -100,20 +100,9 @@ @implementation Branch
...
-    
-    if (!currInstance) {
-        [Branch initInstance];
-    }
-    
-    return currInstance;
-}
-
 + (Branch *)getInstance {
     if (!currInstance) {
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: can we add some key validation and warning code? I want to make sure forgetting the plist step is really obvious and the the logs will tell them what to do

Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: Good point!

**Comment by aaustin** [comment]
:+1:

**Comment by qinweigong** [comment]
I've added more change logs. Please take a peek.

**Comment by aaustin** [comment]
:+1: looks good!

**Pull request merged**